File: mobile-client/android/UseCase/Catalog/Detail/LoadCatalogNodeDetailUseCase.kt
package app.mobiling.client.client.usecase.catalog.detail

import app.mobiling.client.client.contract.catalog.detail.CatalogNodeDetailPayload
import app.mobiling.client.client.data.catalog.detail.CatalogNodeDetailGateway

/**
 * Marketing America Corp. Oleksandr Tishchenko
 */
class LoadCatalogNodeDetailUseCase(
    private val gateway: CatalogNodeDetailGateway,
) {
    suspend operator fun invoke(nodeId: String): CatalogNodeDetailPayload = gateway.loadNodeDetail(nodeId)
}

File: mobile-client/ios/UseCase/Catalog/Detail/LoadCatalogNodeDetailUseCase.swift
import Foundation

// Marketing America Corp. Oleksandr Tishchenko
public struct LoadCatalogNodeDetailUseCase {
    public let gateway: CatalogNodeDetailGateway

    public init(gateway: CatalogNodeDetailGateway) {
        self.gateway = gateway
    }

    public func callAsFunction(nodeId: String) async throws -> CatalogNodeDetailPayload {
        try await gateway.loadNodeDetail(nodeId: nodeId)
    }
}
